modules/extensions/src/views/lang.ts

```diff
@@ -0,0 +1,66 @@
+import { MultiLangText } from 'botpress/sdk'
+import { get } from 'lodash'
+import { createIntl, createIntlCache, IntlShape } from 'react-intl'
+
+import en from '../translations/en.json'
+import fr from '../translations/fr.json'
+
+const defaultLocale = 'en'
+const translations = { fr, en }
+
+let store: any
+let intl: IntlShape
+const cache = createIntlCache()
+let currentLocale = ''
+
+const langInit = () => {
+  const messages = squash(translations[currentLocale])
+  const defaultLang = squash(translations[defaultLocale])
+  for (const key in defaultLang) {
+    if (!messages[key]) {
+      messages[key] = defaultLang[key]
+    }
+  }
+
+  intl = createIntl(
+    {
+      locale: currentLocale,
+      messages,
+      defaultLocale
+    },
+    cache
+  )
+}
+
+const squash = (space, root = {}, path = '') => {
+  for (const [key, value] of Object.entries(space)) {
+    if (typeof value === 'object' && value !== null) {
+      squash(value, root, path + key + '.')
+    } else {
+      root[path + key] = value
+    }
+  }
+  return root
+}
+
+export const lang = {
+  tr: (id: string | MultiLangText, values?: { [variable: string]: any }): string => {
+    if (currentLocale != store.botUILanguage) {
+      currentLocale = store.botUILanguage
+      langInit()
+    }
+
+    if (!id) {
+      return ''
+    }
+
+    if (typeof id === 'object') {
+      return id[currentLocale] || id[defaultLocale] || ''
+    }
+
+    return intl.formatMessage({ id }, values)
+  },
+  set: (thestore: any) => {
+    store = thestore
+  }
+}
```

Comment:
Please check in `common/lang/index.ts`, I had to implement a lightweight translation for prompts. Maybe we can start off that to implement a "lite" version of translations

react-intl adds too many useless stuff and the bundle size is huge. We can use only intl-messageformat to keep the size small. Have a look at this PR for the lightweight implementation: https://github.com/botpress/botpress/pull/3532/files